### DISCUSSION TITLE
Please document the difference between the recovery and release EFIs and how to use recovery

### DISCUSSION THREAD

**Discussion by timkgh** [zbm-dev/zfsbootmenu#577] [Feature Requests]
What is the difference between the recovery and release EFI?

Should the recovery EFI also be available as a boot entry? Documentation shows how to set up another "backup" EFI entry but it's the same EFI image. Is that backup supposed to be the recovery EFI instead?

When is the recovery necessary and the release is not good enough?

Thank you for clarifying the confusion.

**Comment by ahesford** [comment]
Recovery images include [extra tools](https://github.com/zbm-dev/zfsbootmenu/blob/master/etc/zfsbootmenu/recovery.conf.d/recovery.conf). You use the recovery image if you need access to those tools in a pre-boot environment.

The docs recommend a backup EFI entry that is just a copy of the regular image so that you have an option to boot if you accidentally overwrite or otherwise render the primary copy unusable. You can make the recovery image your backup if you like.

**Reply by timkgh** [reply]
I ended up setting the recovery EFI with `zbm.timeout=-1` as the "backup" entry and I hope to never need it 😄

**Comment by classabbyamp** [comment]
now available: https://docs.zfsbootmenu.org/en/v2.3.x/guides/binary-releases.html

**Reply by timkgh** [reply]
Based on the description of what's included, does it make sense for the `release` image to exist? Why not just always use the `recovery` as the only image? Is there a size concern maybe? The difference is only ~15MB.

**Reply by zdykstra** [reply]
Recovery images are 33% larger, with most of that space usage coming from network firmware/drivers. Right now there's no plan to combine them.

**Reply by timkgh** [reply]
Is the extra size a concern that it may not boot on some systems? I'm just curious, that's all and thank you for your patience.

**Reply by zdykstra** [reply]
Given the design choices of the project (Linux kernel + user land), we try to offer the smallest file size possible for space-constrained users. While I imagine most people could just as easily install a 61MB EFI over a 45MB EFI, it's a goal to try to avoid file size bloat.

**Reply by ahesford** [reply]
It's not worth the extra space for most people. If you want to have the tools available, feel free to use that image instead of the standard release.

**Reply by timkgh** [reply]
Makes sense, thank you.
